spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -267,6 +267,43 @@ def receiver.method_missing(*a); end # a poor man's stub...
         it_behaves_like "resets partial mocks cleanly" do
           let(:target) { allow(object) }
         end
+
+        context 'on a class method, from a class with subclasses' do
+          let(:superclass) { Class.new { def self.foo; "foo"; end }}
+          let(:subclass1)  { Class.new(superclass) { def self.foo; super.upcase; end }}
+          let(:subclass2)  { Class.new(superclass) }
```

Comment:
I think the test would be easier to follow if `subclass1` and `subclass2` were better named, e.g. `upcase_subclass` and `no_override_subclass` or similar.